commit 70b18cc4c05398cd06558f4634937a70c2436546
Author: Piotr Dziwinski <piotrdz@gmail.com>
Date:   Sat Jan 18 21:35:32 2014 +0100

    Move engine frame update after game state update
    
    This should finally fix #225

--- a/src/app/app.cpp
+++ b/src/app/app.cpp
@@ -962,15 +962,15 @@
             {
                 LogEvent(event);
 
-                StartPerformanceCounter(PCNT_UPDATE_ENGINE);
-                m_engine->FrameUpdate();
-                StopPerformanceCounter(PCNT_UPDATE_ENGINE);
-
                 m_sound->FrameMove(m_relTime);
 
                 StartPerformanceCounter(PCNT_UPDATE_GAME);
                 m_robotMain->ProcessEvent(event);
                 StopPerformanceCounter(PCNT_UPDATE_GAME);
+
+                StartPerformanceCounter(PCNT_UPDATE_ENGINE);
+                m_engine->FrameUpdate();
+                StopPerformanceCounter(PCNT_UPDATE_ENGINE);
             }
 
             StopPerformanceCounter(PCNT_UPDATE_ALL);
